app-office/texmacs/texmacs-1.99.12.recipe

```diff
@@ -0,0 +1,101 @@
+SUMMARY="A scientific editing platform designed to create beautiful technical documents"
+DESCRIPTION="TeXmacs provides a unified and user friendly framework for editing \
+structured documents with different types of content: text, mathematics, graphics, \
+interactive content, slides, etc.\
+Its rendering engine uses high-quality typesetting algorithms so as to produce \
+professionally looking documents, which can either be printed out or presented from \
+a laptop."
+HOMEPAGE="http://www.texmacs.org/"
+COPYRIGHT="Free Software Foundation"
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="https://github.com/mgubi/texmacs/archive/texmacs-$portVersion-haiku-qt5.tar.gz"
+CHECKSUM_SHA256="607fa4db4676b78780f7af086feec2e2332dc18a371d79fa50d640de6db10b43"
+SOURCE_DIR="texmacs-texmacs-$portVersion-haiku-qt5/src"
+PATCHES="texmacs-$portVersion.patchset"
+ADDITIONAL_FILES="texmacs.rdef.in"
+
+ARCHITECTURES="?x86_gcc2 ?x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	texmacs$secondaryArchSuffix = $portVersion
+	app:TeXmacs
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libiconv$secondaryArchSuffix
+	lib:libQt5Concurrent$secondaryArchSuffix
+	lib:libQt5Core$secondaryArchSuffix
+	lib:libQt5Gui$secondaryArchSuffix
+	lib:libQt5Svg$secondaryArchSuffix
+	lib:libQt5Widgets$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	lib:libguile$secondaryArchSuffix == 17.4.0
+	lib:libfreetype$secondaryArchSuffix
+	lib:libgmp$secondaryArchSuffix
+	lib:libpng16$secondaryArchSuffix
+	lib:libltdl$secondaryArchSuffix
+	lib:libGL$secondaryArchSuffix
+	lib:libintl$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libiconv$secondaryArchSuffix
+	devel:libQt5Concurrent$secondaryArchSuffix
+	devel:libQt5Core$secondaryArchSuffix
+	devel:libQt5Gui$secondaryArchSuffix
+	devel:libQt5Svg$secondaryArchSuffix
+	devel:libQt5Widgets$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	devel:libguile$secondaryArchSuffix == 17.4.0
+	devel:libfreetype$secondaryArchSuffix
+	devel:libgmp$secondaryArchSuffix
+	devel:libltdl$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:autoconf
+	cmd:automake
+	cmd:find
+	cmd:gcc$secondaryArchSuffix
+	cmd:gettext$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	cmd:sed
+	cmd:guile_config$secondaryArchSuffix == 1.8.8
+	cmd:guile$secondaryArchSuffix == 1.8.8
+	"
+
+BUILD()
+{
+	runConfigure ./configure
+	make $jobArgs
+}
+
+INSTALL()
+{
+	make install
+
+	mkdir $appsDir
+	rm $binDir/texmacs
+	mv $binDir/fig2ps $libDir/TeXmacs/bin/fig2ps
+	mv $libDir/TeXmacs/bin/texmacs.bin $appsDir/TeXmacs
```

Comment:
Looks a bit odd to me, for one $libDir/TeXmacs/bin doesn't exist?
Is the location correct? (normally a binary shouldn't go into $libDir?)
Building atm to check

Reply:
In other Unix systems these auxiliary binaries (which are needed internally to convert or to interface to external programs) are put in places like /usr/local/lib/TeXmacs/bin, so I kept the same style. The only difference is that usually fig2ps is installed in /usr/local/bin/fig2ps but I didn't wanted to pollute this place and moved it with the others. On my system the packaging works fine but I do not have a secondary architecture. I'm looking into it, probably there is something wrong in my use of $libDir .

Reply:
suggestions?

Reply:
what is the content of wrapper-script in your machine?

Reply:
I think I made a mistake here, and needed to use $libExecDir which according to https://github.com/haikuports/haikuports/wiki/HaikuPorter-BuildRecipes is the place where one can put the internal executables of the port.  I changed the recipe. Can you try again? My vm is 64 bit and I do not have a setup with a secondary architecture.

Reply:
Cleaning the build and will check again